Update CSSProperties.json with 'initial' and 'animation-type' fields

weinig opened the pull request:
#### d8bfd1cd3e4416467ed95015b220edc6fc28875b
<pre>
Update CSSProperties.json with &apos;initial&apos; and &apos;animation-type&apos; fields
<a href="https://bugs.webkit.org/show_bug.cgi?id=287823">https://bugs.webkit.org/show_bug.cgi?id=287823</a>

Reviewed by Tim Nguyen.

Updates style properties in CSSProperties.json with &quot;initial&quot; and
&quot;animation-type&quot; fields taken from specification property definition
tables. These currently don&apos;t have any effect on code generation
but are validated by process-css-properties.py. They will be used
for code generation in follow up changes.

To increase clarity, a few fields have been renamed:

     &quot;animatable&quot; -&gt; &quot;animation-property&quot; (also moved to codegen-properties)

     &quot;getter&quot; -&gt; &quot;render-style-getter&quot;
     &quot;setter&quot; -&gt; &quot;render-style-setter&quot;
     &quot;initial&quot; -&gt; &quot;render-style-initial&quot;

     &quot;custom&quot; -&gt; &quot;style-builder-custom&quot;
     &quot;converter&quot; -&gt; &quot;style-builder-converter&quot;
     &quot;conditional-converter&quot; -&gt; &quot;style-builder-conditional-converter&quot;
     &quot;name-for-methods&quot; -&gt; &quot;style-builder-name-for-methods&quot;

Additionally, instead of using a &quot;comment&quot; field, properties that
use &quot;top-priority&quot; now must specify a &quot;top-priority-reason&quot; field.

With that use of comment removed, now each shematized dictionary
has an implicitly allowed &quot;comment&quot; field for general commenting,
and every field &quot;foo&quot; has a sibling field &quot;foo-comment&quot; to allow
commenting on that specific field. This is used, for example, to
add explanations when a field doesn&apos;t match the current spec.

While going through the list, URLs and keyword lists were updated
where out of date.

* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/process-css-properties.py:
* Tools/Scripts/webkitpy/style/checkers/jsonchecker.py:

Canonical link: <a href="https://commits.webkit.org/290607@main">https://commits.webkit.org/290607@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/370bdab18cb0523eae6ac00ae2e1b9137e1861ba

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/90380 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/9910 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/45307 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/95386 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/41156 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/92432 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/10297 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/18228 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/69587 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/27155 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/93381 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/7893 "Found 1 new test failure: fast/forms/ios/focus-input-in-fixed.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/81991 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/49931 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/89881 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/7621 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/36354 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/40288 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/77965 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/37424 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/97218 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/17568 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/12928 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/78567 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/17825 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/77814 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/77787 "Found 1 new API test failure: /WebKitGTK/TestCookieManager:/webkit/WebKitCookieManager/replace-get-all-cookies (failure)") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/19245 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/22239 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/20852 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/10835 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/17578 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/22914 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/17319 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/20771 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/19103 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/css/CSSProperties.json:
```diff
@@ -1240,28 +1347,50 @@
         "text-autospace": {
+            "animation-type": "discrete",
             "inherited": true,
+            "initial": "normal",
```
This doesn't match `initialTextAutospace` (see https://github.com/WebKit/WebKit/pull/40309 ).

weinig replied:
Fixed. Thanks.

nt1m commented on Tools/Scripts/webkitpy/style/checkers/jsonchecker.py:
```diff
@@ -347,11 +347,15 @@ def check_property(self, property_name, value):
         keys_and_validators = {
             '*': self.validate_comment,
             'animatable': self.validate_boolean,
```
It would be nice to rename this to 'animation-property' or something.

weinig replied:
Good idea. Done.

nt1m reviewed (approved):
Seems fine, I haven't been able to check that all the `animation-type` and `initial` fields are correct and match what's implemented, but I trust that you've done so, or are planning to get to it in the process :)

weinig committed:
Update CSSProperties.json with 'initial' and 'animation-type' fields
https://bugs.webkit.org/show_bug.cgi?id=287823

Reviewed by Tim Nguyen.

Updates style properties in CSSProperties.json with "initial" and
"animation-type" fields taken from specification property definition
tables. These currently don't have any effect on code generation
but are validated by process-css-properties.py. They will be used
for code generation in follow up changes.

To increase clarity, a few fields have been renamed:

     "animatable" -> "animation-property" (also moved to codegen-properties)

     "getter" -> "render-style-getter"
     "setter" -> "render-style-setter"
     "initial" -> "render-style-initial"

     "custom" -> "style-builder-custom"
     "converter" -> "style-builder-converter"
     "conditional-converter" -> "style-builder-conditional-converter"
     "name-for-methods" -> "style-builder-name-for-methods"

Additionally, instead of using a "comment" field, properties that
use "top-priority" now must specify a "top-priority-reason" field.

With that use of comment removed, now each shematized dictionary
has an implicitly allowed "comment" field for general commenting,
and every field "foo" has a sibling field "foo-comment" to allow
commenting on that specific field. This is used, for example, to
add explanations when a field doesn't match the current spec.

While going through the list, URLs and keyword lists were updated
where out of date.

* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/process-css-properties.py:
* Tools/Scripts/webkitpy/style/checkers/jsonchecker.py:

Canonical link: https://commits.webkit.org/290607@main
